physical/oci_objectstorage/oci_objectstorage_ha.go

```diff
@@ -0,0 +1,554 @@
+			break OUTER
+		}
+
+		// Wait for the timer
+		<-ticker.C
+
+		lockRecordCache := loadLockRecordCache(l)
+		if (lockRecordCache == nil) ||
+			(lockRecordCache.lockRecord == nil) ||
+			(lockRecordCache.lockRecord.Identity != l.identity) ||
+			(time.Now().UTC().Sub(lockRecordCache.time) > l.ttl) {
+			l.backend.logger.Debug("physical/oci: WatchLock: Lock record cache is nil, stale or does not belong to self.")
+			break OUTER
+		}
+
+		lockRecord, _, err := l.get(context.Background())
+		if err != nil {
+			retries++
+			l.backend.logger.Debug("physical/oci: WatchLock: Failed to get lock data from object storage. Retrying..")
+			metrics.SetGauge(metricHaWatchLockRetriable, 1)
+			continue
+		}
+		// Verify the Identity is the same and lock was renewed before its TTL
+		if (lockRecord == nil) || (lockRecord.Identity != l.identity) {
+			l.backend.logger.Debug("physical/oci: WatchLock: Lock record cache is nil or does not belong to self.")
+			break OUTER
+		}
+
+		// reset retries counter on success
+		retries = 0
+		l.backend.logger.Debug("physical/oci: WatchLock() successful")
+		metrics.SetGauge(metricHaWatchLockRetriable, 0)
+	}
+
+	l.stopLock.Lock()
+	defer l.stopLock.Unlock()
+	if !l.stopped {
+		l.stopped = true
+		l.backend.logger.Debug("physical/oci: Closing the stop channel to give up leadership.")
+		close(l.stopCh)
+	}
+}
+
+func (l *Lock) Unlock() error {
+	l.backend.logger.Debug("physical/oci: Unlock() called")
+	defer metrics.MeasureSince(metricLockUnlock, time.Now().UTC())
+
+	l.internalLock.Lock()
+	defer l.internalLock.Unlock()
+	if !l.held {
+		return nil
+	}
+
+	// Stop any existing locking or renewal attempts
+	l.stopLock.Lock()
+	if !l.stopped {
+		l.stopped = true
+		close(l.stopCh)
+	}
+	l.stopLock.Unlock()
+
+	// We are no longer holding the lock
+	l.held = false
+
+	// Get current lock record
+	currentLockRecord, etag, err := l.get(context.Background())
+	if err != nil {
+		return errwrap.Wrapf("error reading lock record: {{err}}", err)
+	}
+
+	if currentLockRecord != nil && currentLockRecord.Identity == l.identity {
+
+		defer metrics.MeasureSince(metricDeleteHa, time.Now())
+		opcClientRequestId, err := uuid.GenerateUUID()
+		if err != nil {
+			l.backend.logger.Debug("physical/oci: Unlock: error generating UUID")
+			opcClientRequestId = ""
+		}
+		l.backend.logger.Debug("physical/oci: Unlock", "opc-client-request-id", opcClientRequestId)
+		request := objectstorage.DeleteObjectRequest{
+			NamespaceName:      &l.backend.namespaceName,
+			BucketName:         &l.backend.lockBucketName,
+			ObjectName:         &l.key,
+			IfMatch:            &etag,
+			OpcClientRequestId: &opcClientRequestId,
+		}
+
+		response, err := l.backend.client.DeleteObject(context.Background(), request)
+		l.backend.logRequest("deleteHA", response.RawResponse, response.OpcClientRequestId, response.OpcRequestId, err)
```

Comment:
Same comment as in oci_objectstorage.go re: it probably being better for this function to take a Request object and handle nil checking there.